### ISSUE TITLE
DFBDF family of solvers with static arrays fails

### ISSUE THREAD

**Issue by davidschlegel** [SciML/OrdinaryDiffEq.jl#1814]
If I write an in-place function for a DAE problem
```
using StaticArrays
using DifferentialEquations
g(du, u, p, t) = du .^2 - conj.(u)
u0 = @SVector randn(ComplexF64, 10)
du0 = @SVector randn(ComplexF64, 10)
prob = DAEProblem(g, du0, u0, (0.0, 10.0))
sol = solve(prob, DFBDF())
```
I get the error `ERROR: MethodError: no method matching eps(::ComplexF64)`

Is there a possible fix?

**Comment by davidschlegel** [comment]
Yes! Indeed for other solvers it works. Thanks. So it is an issue with `DFBDF`.

**Comment by davidschlegel** [comment]
> If I write an in-place function for a DAE problem
> 
> ```
> using StaticArrays
> using DifferentialEquations
> g(du, u, p, t) = du .^2 - conj.(u)
> u0 = @SVector randn(ComplexF64, 10)
> du0 = @SVector randn(ComplexF64, 10)
> prob = DAEProblem(g, du0, u0, (0.0, 10.0))
> sol = solve(prob, DFBDF())
> ```
> 
> I get the error `ERROR: MethodError: no method matching eps(::ComplexF64)`
> 
> Is there a possible fix?

The issue with `eps` not working for complex numbers was already addressed in this pull request: [#58](https://github.com/SciML/LinearSolve.jl/pull/58) in `LinearSolve.jl` and in the yet un-merged pull request [#21858](https://github.com/JuliaLang/julia/pull/21858) in the julia base. Not sure why this still gives an error.

**Comment by ChrisRackauckas** [comment]
Thanks for the report and a clear MWE. I just had to get back from traveling to dig in. It's solved with https://github.com/SciML/OrdinaryDiffEq.jl/pull/1821 and the update to SimpleNonlinearSolve. Will tag and merge today.
